master/internal/api_experiment.go

```diff
@@ -903,6 +938,23 @@ func (a *apiServer) PatchExperiment(
 		}
 	}
 
+	if exp.GroupId != req.Experiment.GroupId && maskContains(mask, "group_id") {
+		if req.Experiment.GroupId != nil {
+			var destGroup *projectv1.ExperimentGroup
+			destGroup, err = a.GetExperimentGroupByID(*req.Experiment.GroupId)
+			if err != nil {
+				return nil, err
+			}
+			madeChanges = true
+			exp.GroupId = &destGroup.Id
+			exp.GroupName = &destGroup.Name
+		} else if len(mask) > 0 {
```

Comment:
Actually, as implemented `maskContains(mask, "group_id")` will return true when `mask` is an empty array (`len(mask) == 0`). I did that so that api calls that didn't include an updateMask would still work as normal. Let me know if you have a better idea for how to handle this.